### DISCUSSION TITLE
Migrate to new application structure and guidelines

### DISCUSSION THREAD

**Discussion by ilmotta** [status-im/status-legacy#16661] [General]
# New application structure and guidelines

This issue will be used to keep track of the work left to be done to adapt the codebase to the new structure and guidelines. See [New Guidelines](https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md) for more information.

**THIS ISSUE IS A WORK IN PROGRESS**, and it'll probably be in this state for a very long time.

- **Prefer to refactor code that will be used in the new Status mobile app**. It's fine to delete dead code (namespaces, functions, events, subs, etc) in the old namespaces, but don't waste time refactoring them if you're not sure the code will be reused in the new mobile app.

## Tasks

- Before working on any of the following tasks, please, **create an issue**, so we reduce the chance of developers doing duplicated work.
- When creating a PR to solve any task, remember NOT to automatically close this issue, and to go back here and mark the task as *done* after merging.
- Please, update the description of this PR to update the list of tasks instead of leaving comments.

#### Misc

- [ ] Remove unused, but registered effects in `status-im.http.core`, `:http-get` and `:http-get-n`.
- [ ] As of 2022-12-15, there are 60 namespaces in `status-im2` directly referencing `quo2.components`. They should use `quo2.core` instead.
- [ ] As of 2022-12-15, there are 18 namespaces in `status-im2` using the old `status-im.i18n.i18n` namespace. They should use `i18n.i18n` instead.
- [ ] As of 2022-12-15, there are 21 namespaces in `status-im2` using old `quo.*` namespaces. For example, `quo.react`, `quo.previews`, etc.
- [ ] Remove `status-im.utils.fx` namespace with helper functions `<sub`, `>evt` in favor of `utils.re-frame` (#14569)

### Useful scripts to evaluate our progress

```shell
# Find files requiring old namespaces in the new src/status-im2 directory.
grep -r \
  -e '\[quo\.' \
  -e '\[status-im\.i18n\.i18n' \
  -e '\[quo2.components\.' \
  -e 'status-im\.utils\.fx' \
  src/status_im2
```

**Comment by flexsurfer** [comment]
`Misc` is good to have, but others I'm not sure, because there will be new designs, and better to move when we'll be  implementing new designs, its not only about visual changes, the entire functionality of the browser and wallet will be new

**Comment by ilmotta** [comment]
> `Misc` is good to have, but others I'm not sure, because there will be new designs, and better to move when we'll be implementing new designs, its not only about visual changes, the entire functionality of the browser and wallet will be new

Indeed @flexsurfer, I wonder the same. I'll update the description to keep the Misc section only.

**Comment by cammellos** [comment]
moving this to a discussion
